accelerator/apis.c

```diff
@@ -237,25 +237,38 @@ status_t api_mam_send_message(const iota_config_t* const tangle, const iota_clie
   const bool last_packet = true;
   bundle_transactions_t* bundle = NULL;
   bundle_transactions_new(&bundle);
+  tryte_t* prng = NULL;
   tryte_t channel_id[MAM_CHANNEL_ID_TRYTE_SIZE];
   trit_t msg_id[MAM_MSG_ID_SIZE];
   ta_send_mam_req_t* req = send_mam_req_new();
   ta_send_mam_res_t* res = send_mam_res_new();
 
-  // Creating MAM API
-  if (mam_api_init(&mam, (tryte_t*)SEED)) {
+  ret = send_mam_req_deserialize(payload, req);
+  if (ret) {
     ret = SC_MAM_FAILED_INIT;
     goto done;
   }
 
-  ret = send_mam_req_deserialize(payload, req);
-  if (ret) {
+  // Creating MAM API
+  prng = (req->prng[0]) ? req->prng : SEED;
+  if ((ret = mam_api_init(&mam, prng)) != RC_OK) {
```

Comment:
`if(mam_api_init(&mam, prng) != RC_OK) ` or `if(mam_api_init(&mam, prng))` would be fine, since `status_t` and `retcode_t` is different.